pkg/sync/v14/data/RoleBinding.rbac.authorization.k8s.io/kube-system/metrics-server-auth-reader.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: RoleBinding
+metadata:
+  name: metrics-server-auth-reader
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
```

Comment:
as a one-off please can you manually check the apiserver audit logs to prove that the metrics server is pulling the extension-apiserver-authentication configmap?

Reply:
I can see in the metrics-server restarts on startup that it attempts to read it, but it was not yet available:

```
W1127 05:39:59.892087       1 authentication.go:220] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
F1127 05:39:59.892129       1 heapster.go:97] Could not create the API server: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:openshift-monitoring:metrics-server" cannot get configmaps in the namespace "kube-system": no RBAC policy matched
```

Reply:
thanks for checking - although it's strange that this error happens at all - do the sync pod logs show that the pod is being created, e.g. before the RBAC is?  If so that's a minor error.  Could you investigate - perhaps the filter configuration in `func (s *sync) writeDB()` needs to be adjusted to guarantee RBAC earlier in the sync run?

Reply:
yes, looked at the logs and Deployment is being created just before RoleBinding. I can see deployment and RoleBinding are in the "other resources" so that's why it is not guaranteed, could be an alphabetic order? https://github.com/openshift/openshift-azure/blob/0c718f237b8ea05224abd984bb6e4c26b15d345b/pkg/sync/v14/sync.go#L328-L392

Reply:
@jim-minter should we include this in this PR then https://github.com/openshift/openshift-azure/pull/2099/commits/ac8afce4f38f31861a80bd9e88e450ce14c4a876?
(I'll squash new commits before making this PR ready to review)